x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java

```diff
@@ -0,0 +1,288 @@
+            expectThrows(SearchPhaseExecutionException.class, () -> client().search(request).actionGet());
+        }
+
+        // Allow the searchable snapshots to be finally mounted
+        unblockNode(repositoryName, dataNodeHoldingSearchableSnapshot);
+        waitUntilRecoveryIsDone(searchableSnapshotIndexOutsideSearchRange);
+        ensureGreen(searchableSnapshotIndexOutsideSearchRange);
+
+        final IndexMetadata updatedIndexMetadata = getIndexMetadata(searchableSnapshotIndexOutsideSearchRange);
+        final IndexLongFieldRange updatedTimestampMillisRange = updatedIndexMetadata.getTimestampMillisRange();
+        assertThat(updatedTimestampMillisRange.isComplete(), equalTo(true));
+        assertThat(updatedTimestampMillisRange, not(sameInstance(IndexLongFieldRange.EMPTY)));
+        assertThat(updatedTimestampMillisRange.getMin(), greaterThanOrEqualTo(Instant.parse("2020-11-26T00:00:00Z").getMillis()));
+        assertThat(updatedTimestampMillisRange.getMax(), lessThanOrEqualTo(Instant.parse("2020-11-27T00:00:00Z").getMillis()));
+        assertThat(indicesService.getTimestampFieldType(updatedIndexMetadata.getIndex()), notNullValue());
+
+        // Stop the node holding the searchable snapshots, and since we defined
+        // the index allocation criteria to require the searchable snapshot
+        // index to be allocated in that node, the shards should remain unassigned
+        internalCluster().stopNode(dataNodeHoldingSearchableSnapshot);
+        waitUntilAllShardsAreUnassigned(updatedIndexMetadata.getIndex());
+
+        if (includeIndexCoveringSearchRangeInSearchRequest) {
+            SearchResponse newSearchResponse = client().search(request).actionGet();
+
+            assertThat(newSearchResponse.getSkippedShards(), equalTo(indexOutsideSearchRangeShardCount));
+            assertThat(newSearchResponse.getSuccessfulShards(), equalTo(totalShards));
+            assertThat(newSearchResponse.getFailedShards(), equalTo(0));
+            assertThat(newSearchResponse.getTotalShards(), equalTo(totalShards));
+        } else {
+            if (indexOutsideSearchRangeShardCount == 1) {
+                expectThrows(SearchPhaseExecutionException.class, () -> client().search(request).actionGet());
+            } else {
+                SearchResponse newSearchResponse = client().search(request).actionGet();
+                // When all shards are skipped, at least one of them should be queried in order to
+                // provide a proper search response.
+                assertThat(newSearchResponse.getSkippedShards(), equalTo(indexOutsideSearchRangeShardCount - 1));
+                assertThat(newSearchResponse.getSuccessfulShards(), equalTo(indexOutsideSearchRangeShardCount - 1));
+                assertThat(newSearchResponse.getFailedShards(), equalTo(1));
+                assertThat(newSearchResponse.getTotalShards(), equalTo(indexOutsideSearchRangeShardCount));
+            }
+        }
```

Comment:
Can we also do a search against a time-range that covers the unavailable searchable snapshot indices to verify that we in that case get errors out.

Reply:
That's what we do in https://github.com/elastic/elasticsearch/pull/65583/files#diff-70b2205b5049ff2a1de6ce21392782676a7aea41ae71e35269e1ee9d3a734ba2R164
In the first search we verify that when the searchable snapshot indices haven't published its min/max timestamp the search fails.

Reply:
I think that is different. On that line, we verify that when we have no min/max and the shard is unavailable, we error out when searching for data outside the range.

What I am asking for here is to verify that when we have min/max available on the coordinator, we error out if searching for a time-range covering unavailable data. I.e., that the can-match passes the search on to the query phase for the shard.